src/core/edgeNetwork/injectSendEdgeNetworkRequest.js

```diff
@@ -16,6 +16,15 @@ import { createCallbackAggregator, noop } from "../../utils/index.js";
 import mergeLifecycleResponses from "./mergeLifecycleResponses.js";
 import handleRequestFailure from "./handleRequestFailure.js";
 
+const isDemdexBlockedError = (error, request) => {
+  return (
+    request.getUseIdThirdPartyDomain() &&
+    (error.name === "TypeError" || // Failed to fetch
```

Comment:
I would make TypeError and NetworkError constants.

Also I would make an utility metod for this condition `(error.name === "TypeError" ||  error.name === "NetworkError" ||  error.status === 0)`